### PR TITLE
added pino

### PR DESCRIPTION
**Pull request by psnider**
case 1. Add a new type definition for pino.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

### CODE REVIEW

File: pino/pino.d.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        safe?: boolean
+        serializers?: Serializers
+    }
+
+
+    interface Pino {
+        (): Logger
+        (options: LoggerOptions): Logger
+        (stream: stream.Writable): Logger
+        (options: LoggerOptions, stream: stream.Writable): Logger
```
Comment (mhegazy) [inline pino/pino.d.ts]: this should be written as:

```ts
(optionsOrStream?: LoggerOptions | stream.Writable): Logger
(options: LoggerOptions, stream: stream.Writable): Logger
```

File: pino/pino.d.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        warn(obj: Object): void
+        info(msg: string, ...args : any[]): void
+        info(obj: Object, msg: string, ...args : any[]): void
+        info(obj: Object): void
+        debug(msg: string, ...args : any[]): void
+        debug(obj: Object, msg: string, ...args : any[]): void
+        debug(obj: Object): void
+        trace(msg: string, ...args : any[]): void
+        trace(obj: Object, msg: string, ...args : any[]): void
+        trace(obj: Object): void
```
Comment (mhegazy) [inline pino/pino.d.ts]: missing `stringify`

File: pino/pino.d.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        (options: LoggerOptions): Logger
+        (stream: stream.Writable): Logger
+        (options: LoggerOptions, stream: stream.Writable): Logger
+    }
+
+
+
+    interface Logger {
+        child(bindings: Object): this
+        level: string   // 'fatal' | 'error' | 'warn' | 'info' | 'debug' | 'trace'
```
Comment (mhegazy) [inline pino/pino.d.ts]: use string literal types:

```ts
type Levels = 'fatal' | 'error' | 'warn' | 'info' | 'debug' | 'trace'
```

File: pino/pino.d.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        fatal(msg: string, ...args : any[]): void
+        fatal(obj: Object, msg: string, ...args : any[]): void
```
Comment (mhegazy) [inline pino/pino.d.ts]: should be 
```ts
fatal(msg: string, ...args : any[]): void
fatal(obj: {}, msg?: string, ...args : any[]): void
```

File: pino/pino.d.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        (): Logger
+        (options: LoggerOptions): Logger
+        (stream: stream.Writable): Logger
+        (options: LoggerOptions, stream: stream.Writable): Logger
+    }
+
+
+
+    interface Logger {
+        child(bindings: Object): this
```
Comment (mhegazy) [inline pino/pino.d.ts]: Please do not use `Object`. either `{}` or `any` depending on the context. here `{}` would be the correct fit.

File: pino/pino.d.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+declare module 'pino' {
+
+    import stream = require('stream')
+
+    interface Serializers {
+        req(req: Object): Object
+        res(res: Object): Object
+    }
+
```
Comment (mhegazy) [inline pino/pino.d.ts]: extra empty lines. there are multiple of these.

File: pino/pino.d.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+/// <reference path="../node/node.d.ts"/>   // only for require()
+
+declare module 'pino' {
+
+    import stream = require('stream')
+
+    interface Serializers {
+        req(req: Object): Object
```
Comment (mhegazy) [inline pino/pino.d.ts]: This should return `any`.

File: pino/pino.d.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        res(res: Object): Object
+    }
```
Comment (mhegazy) [inline pino/pino.d.ts]: missing `err` function.

File: pino/pino.d.ts (hunk 7 above)
Comment (mhegazy) [inline pino/pino.d.ts]: actually not. this has a well defined shape see https://github.com/mcollina/pino#pinostdserializersreq

File: pino/pino.d.ts (hunk 2 above)
Comment (psnider) [inline pino/pino.d.ts]: I believe stringify() is not part of the public interface.
It is set internally depending upon LoggerOptions.safe

File: pino/pino.d.ts (hunk 5 above)
Comment (psnider) [inline pino/pino.d.ts]: I didn't like using Object, but I was unaware I could use {}.

**Pull request merged**